src/rdkafka_broker.c

```diff
@@ -3940,14 +3958,30 @@ static int rd_kafka_broker_fetch_toppars (rd_kafka_broker_t *rkb, rd_ts_t now) {
         if (unlikely(rkb->rkb_active_toppar_cnt == 0))
                 return 0;
 
-	rkbuf = rd_kafka_buf_new_request(
+        rkbuf = rd_kafka_buf_new_request(
                 rkb, RD_KAFKAP_Fetch, 1,
-                /* ReplicaId+MaxWaitTime+MinBytes+TopicCnt */
-                4+4+4+4+
-                /* N x PartCnt+Partition+FetchOffset+MaxBytes+?TopicNameLen?*/
-                (rkb->rkb_active_toppar_cnt * (4+4+8+4+40)));
-
-        if (rkb->rkb_features & RD_KAFKA_FEATURE_MSGVER2)
+                /* ReplicaId+MaxWaitTime+MinBytes+MaxBytes+IsolationLevel+
+                 * SessionId+SessionEpoch+TopicCnt
```

Comment:
this actually fixes a bug in the existing code - IsolationLevel is not accounted for yet `RD_KAFKAP_READ_UNCOMMITTED` is written. furthermore, this wasn't fixed in #2377.

Reply:
sorry for the long delay.
@edenhill is this PR still relevant or are there plans to upgrade to full v10 protocol support ?
@mhowlett is it just a rebase that i need to do or there something more specific w.r.t syncing to your work?